models/DataObject/Classificationstore/Service.php

```diff
@@ -22,17 +22,34 @@
 class Service
 {
     /**
-     * @param KeyConfig $keyConfig
+     * @var array Used for storing definitions
+     */
+    protected static $definitionsCache = [];
+
+    /**
+     * @param KeyConfig|KeyGroupRelation $keyConfig
      *
      * @return DataObject\ClassDefinition\Data
      */
     public static function getFieldDefinitionFromKeyConfig($keyConfig)
     {
+        if ($keyConfig instanceof KeyConfig) {
+            $cacheId = $keyConfig->getId();
+        }
+
+        if ($keyConfig instanceof KeyGroupRelation) {
+            $cacheId = $keyConfig->getGroupId() . "_" . $keyConfig->getKeyId();
+        }
```

Comment:
If $keyConfig is no KeyConfig or KeyGroupRelation, $cacheId is not defined.

```suggestion
        if ($keyConfig instanceof KeyConfig) {
            $cacheId = $keyConfig->getId();
        } elseif ($keyConfig instanceof KeyGroupRelation) {
            $cacheId = $keyConfig->getGroupId() . "_" . $keyConfig->getKeyId();
        } else {
            throw new \Exception('$keyConfig should be KeyConfig or KeyGroupRelation');
        }
```

Reply:
As Pimcore now requires PHP 7.2, maybe we can just typehint this?

Reply:
PHP can only one class as typehint, right?

Reply:
@weisswurstkanone is typehinting okay, or should we go with if/else?

Reply:
@blankse you're right, thanks for the tip!